docs/release-process-deep-dive.md

```diff
@@ -0,0 +1,596 @@
+
+> [!WARNING]
+> The `createrepo` tool is executed inside a [Docker container](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/script/createrepo.sh) for [package management reasons](https://github.com/cli/cli/pull/2856) that may no longer be true.
+
+#### Debian
+
+The `.deb` files uplaoded by the [`linux`](#linux) job are iterated per Debian release (see warning below), using [`reprepro`](https://manpages.debian.org/bookworm/reprepro/reprepro.1.en.html) which produces a directory and file structure for a Debian package repository. The [`./script/distributions`](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/script/distributions) `SignWith` lines indicate the GPG Key ID that `reprepro` should use to sign packages in the created repository. The generated directories are then copied to the site repository.
+
+> [!WARNING]
+> There is a note that we should remove [legacy distributions from our list](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/.github/workflows/deployment.yml#L329-L332) but no indication of when that would happen.
+
+### Attest Artifacts
+
+Attestations are created for each of the release artifacts. For an example see: https://github.com/cli/cli/attestations/4920729
+
+### Publish Release
+
+After all release artifacts have been created, and signed, there are a number of steps taken to make them available to our users.
+
+#### GitHub Release
+
+`gh release create` is invoked to create a new release on GitHub, attaching all the archives, packages and installers, plus a checksum file to allow `gh` users to validate the attached artifacts. The artifact file names are provided to `gh release create` along with a label, as per the command `--help`:
+
+```
+Upload a release asset with a display label
+$ gh release create v1.2.3 '/path/to/asset.zip#My display label'
+```
+
+> [!NOTE]
+> TODO: Further Investigation required on why they are labelled this way.
+
+#### Site
+
+In previous steps, a git commit was made for the manual, and files had moved into place for the RPM and Debian package repositories. The package repository structure is commited and pushed, which kicks off a deployment workflow in site repository.
+
+> [!NOTE]
+> TODO: Further Documentation required on cleaning up the package repository.
+
+#### Homebrew Formula
+
+Using [`mislav/bump-homebrew-formula-action`](https://github.com/mislav/bump-homebrew-formula-action), a PR for the `gh` [`homebrew-core` formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/g/gh.rb) is created. The fork repository is currenty owned by `williammartin` as PRs are [not accepted from organizations.](https://github.com/cli/cli/pull/7953)
```

Comment:
Unsure how much more info you want to capture here about new releases only getting picked up based on a scheduled workflow once this PR is merged. take with a grain of 🧂

Reply:
Probably worthwhile since the question has come up before.